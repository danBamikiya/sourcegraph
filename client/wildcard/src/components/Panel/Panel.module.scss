.panel {
    position: fixed;
    padding: 0.3rem 0.5rem;

    background-color: var(--color-bg-1);
}

.bottom {
    border-top: 2px solid var(--border-color-2);
    min-height: 10rem;
    width: 100%;
    bottom: 0;
    left: 0;
}

.side {
    border-left: 2px solid var(--border-color-2);
    min-width: 10rem;
    height: 100%;
    top: 0;
    right: 0;
}

.dismiss-button {
    color: var(--icon-color);
}
