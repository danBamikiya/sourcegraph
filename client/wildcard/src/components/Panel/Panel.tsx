import classNames from 'classnames'
import CloseIcon from 'mdi-react/CloseIcon'
import React from 'react'

import styles from './Panel.module.scss'

interface PanelProps {
    position: 'bottom' | 'side'
    onDismiss: () => void
}

export const Panel: React.FunctionComponent<PanelProps> = ({ children, position, onDismiss }) => (
    <div className={classNames(styles.panel, styles[position])}>
        <div className={classNames('d-flex', position === 'bottom' ? 'justify-content-end' : 'justify-content-start')}>
            <button
                type="button"
                onClick={onDismiss}
                className={classNames('btn btn-icon', styles.dismissButton)}
                title="Close panel"
                data-tooltip="Close panel"
                data-placement="left"
                data-testid="panel-onDismiss-button"
            >
                <CloseIcon className="icon-inline" />
            </button>
        </div>
        {children}
    </div>
)
