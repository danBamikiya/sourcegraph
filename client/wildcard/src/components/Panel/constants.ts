export const PANEL_POSITION = ['bottom', 'side'] as const
