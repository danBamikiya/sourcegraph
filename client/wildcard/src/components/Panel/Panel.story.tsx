import { select } from '@storybook/addon-knobs'
import { DecoratorFn, Meta, Story } from '@storybook/react'
import React from 'react'

import { BrandedStory } from '@sourcegraph/branded/src/components/BrandedStory'
import webStyles from '@sourcegraph/web/src/SourcegraphWebApp.scss'

import { PANEL_POSITION } from './constants'
import { Panel } from './Panel'

const decorator: DecoratorFn = story => <BrandedStory styles={webStyles}>{() => <div>{story()}</div>}</BrandedStory>

const config: Meta = {
    title: 'wildcard/Panel',

    decorators: [decorator],

    parameters: {
        component: Panel,
    },
}

export default config

export const PanelExample: Story = () => (
    <Panel onDismiss={() => {}} position={select('Position', PANEL_POSITION, 'bottom')} />
)
