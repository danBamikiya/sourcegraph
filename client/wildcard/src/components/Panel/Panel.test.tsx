import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import React from 'react'
import sinon from 'sinon'

import { Panel } from './Panel'

describe('Panel', () => {
    it('renders correctly positioned at page bottom', () => {
        expect(render(<Panel position="bottom" onDismiss={() => {}} />).baseElement).toMatchSnapshot()
    })

    it('renders correctly positioned at page side', () => {
        expect(render(<Panel position="side" onDismiss={() => {}} />).baseElement).toMatchSnapshot()
    })

    test('calls onDismiss function when dismiss icon is clicked', () => {
        const closePanel = sinon.spy(() => undefined)
        render(<Panel position="bottom" onDismiss={closePanel} />)

        userEvent.click(screen.getByTestId('panel-onDismiss-button'))
        sinon.assert.called(closePanel)
    })
})
