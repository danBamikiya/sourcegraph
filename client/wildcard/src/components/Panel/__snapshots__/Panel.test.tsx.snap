// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel renders correctly positioned at page bottom 1`] = `
<body>
  <div>
    <div
      class="panel bottom"
    >
      <div
        class="d-flex justify-content-end"
      >
        <button
          class="btn btn-icon dismissButton"
          data-placement="left"
          data-testid="panel-onDismiss-button"
          data-tooltip="Close panel"
          title="Close panel"
          type="button"
        >
          <svg
            class="mdi-icon icon-inline"
            fill="currentColor"
            height="24"
            viewBox="0 0 24 24"
            width="24"
          >
            <path
              d="M19,6.41L17.59,5L12,10.59L6.41,5L5,6.41L10.59,12L5,17.59L6.41,19L12,13.41L17.59,19L19,17.59L13.41,12L19,6.41Z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</body>
`;

exports[`Panel renders correctly positioned at page side 1`] = `
<body>
  <div>
    <div
      class="panel side"
    >
      <div
        class="d-flex justify-content-start"
      >
        <button
          class="btn btn-icon dismissButton"
          data-placement="left"
          data-testid="panel-onDismiss-button"
          data-tooltip="Close panel"
          title="Close panel"
          type="button"
        >
          <svg
            class="mdi-icon icon-inline"
            fill="currentColor"
            height="24"
            viewBox="0 0 24 24"
            width="24"
          >
            <path
              d="M19,6.41L17.59,5L12,10.59L6.41,5L5,6.41L10.59,12L5,17.59L6.41,19L12,13.41L17.59,19L19,17.59L13.41,12L19,6.41Z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</body>
`;
